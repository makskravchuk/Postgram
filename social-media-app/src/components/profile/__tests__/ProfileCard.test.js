import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from "react-test-renderer";
import ProfileCard from "../ProfileCard";
import { BrowserRouter } from "react-router-dom";


const userData = {
    id: "0590cd67-eacd-4299-8413-605bd547ea17",
    first_name: "FirstName",
    last_name: "LastName",
    name: "FirstName LastName",
    post_count: 3,
    email: "example@gmail.com",
    bio: "Example Bio",
    username: "username_1",
    avatar: null,
    created: "2025-01-19T17:31:03.310Z",
    updated: "2025-01-20T07:38:47.631Z",
};


test("Test ProfileCard snapshot", () => {
    const ProfileCardDomTree = TestRenderer.create(
        <BrowserRouter>
            <ProfileCard user={userData}/>
        </BrowserRouter>
    ).toJSON();

    expect(ProfileCardDomTree).toMatchSnapshot();
});