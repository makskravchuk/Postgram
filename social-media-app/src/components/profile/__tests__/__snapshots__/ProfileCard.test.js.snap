// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProfileCard snapshot 1`] = `null`;
